### PR TITLE
Fix the l3node

### PR DESCRIPTION
**Pull request by ImJeremyHe**
### This PR:
Fix the L3 node

Currently, when enabling L3 node, we will run an L2 node without enabling the espresso mode.

Also add a simple test for l3 node in smoke-test

### How to test this PR
Run the `./smoke-test.bash`

### CODE REVIEW

File: smoke-test.bash [hunk 1]
```diff
@@ -19,3 +19,4 @@ while true; do
 done
 
 echo "Smoke test succeeded"
+docker compose down
```
Comment (sveitser) [inline smoke-test.bash:21]: Should do this with `trap` so it runs when the test fails somewhere.

Comment (ImJeremyHe) [reply smoke-test.bash:21]: I have thought of it. But I think we should keep the containers running if the test fails so that we can debug conveniently

**Comment by sveitser** [comment]
I think we should add a CI workflow to run the smoke test on the L3. It can be based on the current smoke test. Otherwise if we break the L3 support via espresso-dev-node or by breaking the bash script we won't know.

File: smoke-test.bash [hunk 2]
```diff
@@ -18,4 +18,22 @@ while true; do
   sleep 5
 done
 
-echo "Smoke test succeeded"
+docker compose down
+
+# Testing the l3node
```
Comment (sveitser) [inline smoke-test.bash]: @ImJeremyHe Lets make it a separate github action workflow run and also test that the staker is working. Either set up a matrix to execute the script with 2 configurations or duplicate the workflow and modify it for the L3. I think both are fine.

**Review by sveitser — APPROVED** [review]
LGTM, but wait for CI to pass

**Pull request merged**